src/openassetio-core-c/include/openassetio/c/InfoDictionary.h

```diff
@@ -28,10 +28,6 @@ extern "C" {
 /**
  * Opaque handle type representing a @fqref{InfoDictionary}
  * "InfoDictionary" instance.
- *
- * The ownership semantics of this handle are "owned by client", that
- * is, the caller of the C API is responsible for deallocating using
- * the suite's `dtor` function once the InfoDictionary is no longer in use.
```

Comment:
[plan] Longer-term, we should have a page on "memory management between languages" or similar, were we outline the strategy/best practices.

Reply:
How about an issue to update the `C_API.md` doc? Perhaps once we've proved out the semi-end-to-end workflow?

Reply:
Sounds good to me!

Reply:
Added https://github.com/TheFoundryVisionmongers/OpenAssetIO/issues/359